helix-rest/src/main/java/org/apache/helix/rest/server/service/InstanceServiceImpl.java

```diff
@@ -275,11 +275,17 @@ private StoppableCheck performCustomInstanceCheck(String clusterId, String insta
         _dataAccessor.getChildValues(_dataAccessor.keyBuilder().externalViews(), true);
     Map<String, StoppableCheck> instanceStoppableChecks = new HashMap<>();
     for (String instanceName : instances) {
-      List<String> unHealthyPartitions = InstanceValidationUtil
+      Map<String, List<String>> unHealthyPartitions = InstanceValidationUtil
           .perPartitionHealthCheck(externalViews, allPartitionsHealthOnLiveInstance, instanceName,
               _dataAccessor);
+
+      List <String> unHealthyPartitionsList = new ArrayList<>();
+      for (String partitionName : unHealthyPartitions.keySet()) {
+        for (String reason : unHealthyPartitions.get(partitionName))
+          unHealthyPartitionsList.add(reason.toUpperCase() + ":" + partitionName);
```

Comment:
Coding convention may say that you need the brace brackets here. :)

Reply:
Done.